packages/next/src/client/components/router-reducer/reducers/navigate-reducer.ts

```diff
@@ -176,7 +176,7 @@ function navigateReducer_noPPR(
           flightSegmentPathWithLeadingEmpty,
           currentTree,
           treePatch,
-          url.pathname
+          href
```

Comment:
Looks like this `href` also contains the `hash` is this intentional?

Reply:
Good q - tested this, and confirmed it's ok since the server discards the hash

Reply:
Added this to the test cases in [f694ef9](https://github.com/vercel/next.js/pull/64086/commits/f694ef9a59d992449046a253d19d7f291782a32f)